internal/tool/prepare.go

```diff
@@ -0,0 +1,134 @@
+package tool
+
+import (
+	"bufio"
+	"os"
+	"path"
+	"strings"
+
+	codacy "github.com/codacy/codacy-engine-golang-seed/v6"
+	"github.com/samber/lo"
+)
+
+const sourceConfigFileName = ".semgrep.yaml"
+
+// TODO: should respect cli flag for docs location
+const rulesDefinitionFileName = "/docs/rules.yaml"
+
+func createConfigFile(toolExecution codacy.ToolExecution) (*os.File, error) {
```

Comment:
```suggestion
func newConfigurationFile(toolExecution codacy.ToolExecution) (*os.File, error) {
```

How do you feel about this? I think it's more consistent with our current naming.

Also, lookout for inconsistencies between `configuration` and `config`.

Reply:
I think that's better yes. I will just have to rename the other function which was "writeTmpFile", now it opens, writes & returns the file (which in essence is creating a new configuration file).